pkg/helmexec/exec.go

```diff
@@ -130,6 +130,9 @@ func (helm *execer) AddRepo(name, repository, cafile, certfile, keyfile, usernam
 		return fmt.Errorf("empty field name")
 	}
 	args = append(args, "repo", "add", name, repository)
+	if helm.IsHelm3() && helm.IsVersionAtLeast(3, 3, 2) {
```

Comment:
Recommend adding another method `IsVersionAtMost` and adding `&& helm.IsVersionAtMost(3, 3, 3)` to this if statement. [A PR is being worked](https://github.com/helm/helm/pull/8772) to let the previous idempotent functionality return as long as the name and url are identical to what they were before.

The security issues that Helm worked to fix with this change are valid. [They are listed in the 3.3.2 release notes](https://github.com/helm/helm/releases/tag/v3.3.2)

My golang fu is not strong, but I can work on doing a PR as it seems to be a small change.

Reply:
This would require an addition to the documentation that changing the URL of an existing named repo will cause an error, but IMO that's a good thing.

Reply:
Actually, I'm seeing another issue. I frequently use the helm git plugin and pin to a specific tag or ref in the git repo. Changing that ref would be considered a change in the URL, which would throw the error.

Thoughts?

Reply:
Git ref changing is content changing, generally speaking, thus throwing an error is a valid option. Looks like we need `-force-update` configurable from helmfile to shift a decision to user side.

Reply:
See #1494.